mypy/main.py

```diff
@@ -252,6 +257,89 @@ def infer_python_executable(options: Options,
   Define MYPY_CACHE_DIR to override configuration cache_dir path."""  # type: Final
 
 
+class CapturableArgumentParser(argparse.ArgumentParser):
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        try:
+            self.stdout = kwargs.pop('stdout')
```

Comment:
Maybe you can avoid the try statements by using `kwargs.pop('stdout', sys.stdout)` (and similarly for stderr).